validator_worker/src/main.rs

```diff
@@ -170,33 +178,26 @@ async fn iterate_channels<A: Adapter + 'static>(args: Args<A>) -> Result<(), ()>
         eprintln!("An occurred while processing channels {}", e);
     }
 
-    println!("processed {} channels", channels_size);
     if channels_size >= args.config.max_channels as usize {
         eprintln!(
             "WARNING: channel limit cfg.MAX_CHANNELS={} reached",
             args.config.max_channels
         )
     }
-
     Ok(())
 }
 
 async fn validator_tick<A: Adapter + 'static>(
     adapter: Arc<RwLock<A>>,
     channel: Channel,
     config: &Config,
-    whoami: &str,
+    whoami: &ValidatorId,
 ) -> Result<(), ValidatorWorkerError> {
     let sentry = SentryApi::init(adapter, &channel, &config, true, whoami)?;
-
-    let index = channel
-        .spec
-        .validators
-        .into_iter()
-        .position(|v| v.id == *whoami);
     let duration = Duration::from_secs(config.validator_tick_timeout as u64);
-    match index {
-        Some(0) => {
+
+    match channel.spec.validators.find(whoami.to_owned()) {
```

Comment:
just use a reference, it will make it much cleaner, we don't need owned value for `find`